pkg/controller/statefulset/stateful_pod_control.go

```diff
@@ -108,23 +145,42 @@ func (spc *realStatefulPodControl) UpdateStatefulPod(set *apps.StatefulSet, pod
 				return err
 			}
 		}
+		// if the Pod's PVCs are not consistent with the StatefulSet's PVC deletion policy, update the PVC
+		// and dirty the pod.
+		if match, err := spc.ClaimsMatchDeletionPolicy(set, pod); err != nil {
+			if consistent {
+				// Oops, something has gone badly wrong.
+				spc.recordPodEvent("update", set, pod, err)
+				return err
+			}
+			// Otherwise the error may be due to some other update, and we'll have to reconcile the next time around.
+			// TODO: does the reconcile need to be sped up?
+		} else if !match {
+			if err := spc.UpdatePodClaimForDeletionPolicy(set, pod); err != nil {
```

Comment:
yes, although it gets complicated because if the pod is new as well it may not have the UID necessary for the owner ref.

So +1 to future optimization :-)